Previously watched

nathanielmillard opened the pull request:
## What's this PR do?
-Renames the GenrePage to Movie List
-Recycles this components to create a previously watched page

### How should this be manually tested?
-Navigate to previously watched from the nav menu

### What are the relevant issues/tickets?

#### Screenshots (if appropriate)
![Screen Shot 2021-02-09 at 2 46 48 PM](https://user-images.githubusercontent.com/65516564/107432764-a5ffec80-6ae5-11eb-939f-a27c440f643f.png)

nathanielmillard committed:
Rename genre page to MovieList.

nathanielmillard committed:
Create route for previously wacthed movie to render.

nathanielmillard committed:
Add conditional to check previous movies if nothing is returned on movies to watch search.

nathanielmillard committed:
Add Link to previously watched.

nathanielmillard committed:
Add conditional logic to render previous movies, title, and dynamic class name.

nathanielmillard committed:
Add previous watch gradient colors.

nathanielmillard committed:
Install dependencies.

nathanielmillard committed:
Pull most recent changes.

nathanielmillard committed:
Add in previously watched.

nathanielmillard committed:
Center test on mobile menu.

nathanielmillard committed:
Remove file that made it in with merge.

nathanielmillard committed:
Resolve merge conflicts.

nathanielmillard committed:
Adjust to single quotes inside double for CI warning.

sschipke commented on src/containers/MovieList/MovieList.scss:
```diff
@@ -3,6 +3,11 @@
 }
 
+.previously-watched {
+  background-image: linear-gradient(to right, #FDE997, #00D8FF);
+  scroll-padding-top: 0;
+}
+
```
Per the comp, I believe this background gradient should be #FDE997 to #FF3399
[Hex Values Here](https://docs.google.com/document/d/18rHWoStJw2ZglnNJvR-GAYEimz3G4wb1BwgdejUGQ98)
## Screenshot of comp
![Screen Shot 2021-02-09 at 4 03 48 PM](https://user-images.githubusercontent.com/49662828/107440146-668acd80-6af0-11eb-96f6-842aff830a61.png)

nathanielmillard replied:
Taken care of! Thanks!

sschipke reviewed (changes requested):
### Observations:

- [x] When I look at a specific movie from the "Previously Watched List" the runtime is displaying as `NAN`
![Screen Shot 2021-02-09 at 4 12 37 PM](https://user-images.githubusercontent.com/49662828/107440908-a1d9cc00-6af1-11eb-8083-a39b013225bc.png)

- [x] In the mobile view, the previously watched movies are rendered outside of their container and off the page 
![602305b3cfe20e00078e5fca--blackstories-dev netlify app_previouslywatched(iPhone 6_7_8)](https://user-images.githubusercontent.com/49662828/107441203-16ad0600-6af2-11eb-9f75-d2cb8f0bdadb.png)

- [x] The film information on the "Previously Watched Page" should display only the following (see comp for examples): 
- Movie Poster
- Title
- Date Watched (`date_watched`)
- Who in the group chose the movie (`chosen_by`)

- [x] The movies should be in chronological order by their `date_watched` which is to say, the first movie chosen should be at the top, with the most current movies at the bottom.

sschipke committed:
Show all, Code of Conduct (#41)

* Gchappell (#9)

* Added foreign genre

* added new genres, added genre tags to movies, and added new films

* Added movies, re-added genres that went missing, added movie to prev mov

* Sschpke/desktop touchups (#12)

* Update video close button

* Adjust movie cards to three per line 1280

* Add PR template

* Remove console log

* Make nav title link

* Address touchups for desktop

* Create consistent margin for main image

* Remove unused close icon

* Do not diplsay fd or bd or slctd genre

* Remove unused var

* Fix movie page overview scroll and backdrop

* Fix scrolling issues and nav jumping

* Update favicon

* Display runtime

* Allow routing to specific movie or genre (#24)

* Display runtime and update favicon (#19)

* Gchappell (#9)

* Added foreign genre

* added new genres, added genre tags to movies, and added new films

* Added movies, re-added genres that went missing, added movie to prev mov

* Sschpke/desktop touchups (#12)

* Update video close button

* Adjust movie cards to three per line 1280

* Add PR template

* Remove console log

* Make nav title link

* Address touchups for desktop

* Create consistent margin for main image

* Remove unused close icon

* Do not diplsay fd or bd or slctd genre

* Remove unused var

* Fix movie page overview scroll and backdrop

* Fix scrolling issues and nav jumping

* Update favicon

* Display runtime

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Add NotFound component and logic

* Add redirects file

* Add foreign to genreMap

* Allow routing to specific movie based on URL

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Add initial issue templates

Out of the box GitHub issue templates

* Sschipke/update data 02 04 21 (#26)

* Display runtime and update favicon (#19)

* Gchappell (#9)

* Added foreign genre

* added new genres, added genre tags to movies, and added new films

* Added movies, re-added genres that went missing, added movie to prev mov

* Sschpke/desktop touchups (#12)

* Update video close button

* Adjust movie cards to three per line 1280

* Add PR template

* Remove console log

* Make nav title link

* Address touchups for desktop

* Create consistent margin for main image

* Remove unused close icon

* Do not diplsay fd or bd or slctd genre

* Remove unused var

* Fix movie page overview scroll and backdrop

* Fix scrolling issues and nav jumping

* Update favicon

* Display runtime

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Move PR template to .github dir

* Place Foreign genre in correct order

* Update movie data

* Update Rize poster path

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Mobile View (#25)

* Display runtime and update favicon (#19)

* Gchappell (#9)

* Added foreign genre

* added new genres, added genre tags to movies, and added new films

* Added movies, re-added genres that went missing, added movie to prev mov

* Sschpke/desktop touchups (#12)

* Update video close button

* Adjust movie cards to three per line 1280

* Add PR template

* Remove console log

* Make nav title link

* Address touchups for desktop

* Create consistent margin for main image

* Remove unused close icon

* Do not diplsay fd or bd or slctd genre

* Remove unused var

* Fix movie page overview scroll and backdrop

* Fix scrolling issues and nav jumping

* Update favicon

* Display runtime

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Make website mobile friendly

* Squashed commit of the following:

commit 215485b8572e3a583112d4b655b9205553f2c3c3
Merge: 6f40d5b 0d99718
Author: Scott <swschipke@gmail.com>
Date:   Thu Feb 4 21:26:52 2021 -0700

    Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

commit 0d997184cfceb3e0418fd8976b384c5a6e584571
Author: Scott Schipke <swschipke@gmail.com>
Date:   Thu Feb 4 21:24:18 2021 -0700

    Sschipke/update data 02 04 21 (#26)

    * Display runtime and update favicon (#19)

    * Gchappell (#9)

    * Added foreign genre

    * added new genres, added genre tags to movies, and added new films

    * Added movies, re-added genres that went missing, added movie to prev mov

    * Sschpke/desktop touchups (#12)

    * Update video close button

    * Adjust movie cards to three per line 1280

    * Add PR template

    * Remove console log

    * Make nav title link

    * Address touchups for desktop

    * Create consistent margin for main image

    * Remove unused close icon

    * Do not diplsay fd or bd or slctd genre

    * Remove unused var

    * Fix movie page overview scroll and backdrop

    * Fix scrolling issues and nav jumping

    * Update favicon

    * Display runtime

    Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

    * Move PR template to .github dir

    * Place Foreign genre in correct order

    * Update movie data

    * Update Rize poster path

    Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

commit 6f40d5b51edc8340e6ae0eaa5f5781725b1968c8
Merge: d9ca34a a90da5e
Author: Scott <swschipke@gmail.com>
Date:   Thu Feb 4 20:01:19 2021 -0700

    Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

commit a90da5e94fd67fff999a37e681cfe4597642099e
Author: Scott Schipke <swschipke@gmail.com>
Date:   Thu Feb 4 20:00:47 2021 -0700

    Add initial issue templates

    Out of the box GitHub issue templates

* Update moviepage to delay setcurntmovie til render

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Add Correct Image Files for Mobile View (#28)

* Update with actual mobile images

* Update displaying mobile image

* Using Ref API in React, scroll to the last movie a user looked at (#27)


* Scroll to prev movie on genrepage

* Make website mobile friendly

* Scroll to prev movie on genrepage

Squashed commit of the following:

commit 4baaa3bb07117e70b8ed2ee25d939af5bd27e314
Author: Scott Schipke <swschipke@gmail.com>
Date:   Sun Feb 7 21:17:56 2021 -0700

    Mobile View (#25)

    * Display runtime and update favicon (#19)

    * Gchappell (#9)

    * Added foreign genre

    * added new genres, added genre tags to movies, and added new films

    * Added movies, re-added genres that went missing, added movie to prev mov

    * Sschpke/desktop touchups (#12)

    * Update video close button

    * Adjust movie cards to three per line 1280

    * Add PR template

    * Remove console log

    * Make nav title link

    * Address touchups for desktop

    * Create consistent margin for main image

    * Remove unused close icon

    * Do not diplsay fd or bd or slctd genre

    * Remove unused var

    * Fix movie page overview scroll and backdrop

    * Fix scrolling issues and nav jumping

    * Update favicon

    * Display runtime

    Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

    * Make website mobile friendly

    * Squashed commit of the following:

    commit 215485b8572e3a583112d4b655b9205553f2c3c3
    Merge: 6f40d5b 0d99718
    Author: Scott <swschipke@gmail.com>
    Date:   Thu Feb 4 21:26:52 2021 -0700

        Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

    commit 0d997184cfceb3e0418fd8976b384c5a6e584571
    Author: Scott Schipke <swschipke@gmail.com>
    Date:   Thu Feb 4 21:24:18 2021 -0700

        Sschipke/update data 02 04 21 (#26)

        * Display runtime and update favicon (#19)

        * Gchappell (#9)

        * Added foreign genre

        * added new genres, added genre tags to movies, and added new films

        * Added movies, re-added genres that went missing, added movie to prev mov

        * Sschpke/desktop touchups (#12)

        * Update video close button

        * Adjust movie cards to three per line 1280

        * Add PR template

        * Remove console log

        * Make nav title link

        * Address touchups for desktop

        * Create consistent margin for main image

        * Remove unused close icon

        * Do not diplsay fd or bd or slctd genre

        * Remove unused var

        * Fix movie page overview scroll and backdrop

        * Fix scrolling issues and nav jumping

        * Update favicon

        * Display runtime

        Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

        * Move PR template to .github dir

        * Place Foreign genre in correct order

        * Update movie data

        * Update Rize poster path

        Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

    commit 6f40d5b51edc8340e6ae0eaa5f5781725b1968c8
    Merge: d9ca34a a90da5e
    Author: Scott <swschipke@gmail.com>
    Date:   Thu Feb 4 20:01:19 2021 -0700

        Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

    commit a90da5e94fd67fff999a37e681cfe4597642099e
    Author: Scott Schipke <swschipke@gmail.com>
    Date:   Thu Feb 4 20:00:47 2021 -0700

        Add initial issue templates

        Out of the box GitHub issue templates

    * Update moviepage to delay setcurntmovie til render

    Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Show All Movies, Display Genres as Links on Movie Page, Display Trigger Warnings, Add Code of Conduct Page (#40)

* Fix love and bask id

* Add link and page to show all movies

Squashed commit of the following:

commit 7ee5ec39fd9b8c5da185fa8b975fc26df4e90e11
Author: Scott Schipke <swschipke@gmail.com>
Date:   Mon Feb 8 13:52:29 2021 -0700

    Using Ref API in React, scroll to the last movie a user looked at (#27)

    * Scroll to prev movie on genrepage

    * Make website mobile friendly

    * Scroll to prev movie on genrepage

    Squashed commit of the following:

    commit 4baaa3bb07117e70b8ed2ee25d939af5bd27e314
    Author: Scott Schipke <swschipke@gmail.com>
    Date:   Sun Feb 7 21:17:56 2021 -0700

        Mobile View (#25)

        * Display runtime and update favicon (#19)

        * Gchappell (#9)

        * Added foreign genre

        * added new genres, added genre tags to movies, and added new films

        * Added movies, re-added genres that went missing, added movie to prev mov

        * Sschpke/desktop touchups (#12)

        * Update video close button

        * Adjust movie cards to three per line 1280

        * Add PR template

        * Remove console log

        * Make nav title link

        * Address touchups for desktop

        * Create consistent margin for main image

        * Remove unused close icon

        * Do not diplsay fd or bd or slctd genre

        * Remove unused var

        * Fix movie page overview scroll and backdrop

        * Fix scrolling issues and nav jumping

        * Update favicon

        * Display runtime

        Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

        * Make website mobile friendly

        * Squashed commit of the following:

        commit 215485b8572e3a583112d4b655b9205553f2c3c3
        Merge: 6f40d5b 0d99718
        Author: Scott <swschipke@gmail.com>
        Date:   Thu Feb 4 21:26:52 2021 -0700

            Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

        commit 0d997184cfceb3e0418fd8976b384c5a6e584571
        Author: Scott Schipke <swschipke@gmail.com>
        Date:   Thu Feb 4 21:24:18 2021 -0700

            Sschipke/update data 02 04 21 (#26)

            * Display runtime and update favicon (#19)

            * Gchappell (#9)

            * Added foreign genre

            * added new genres, added genre tags to movies, and added new films

            * Added movies, re-added genres that went missing, added movie to prev mov

            * Sschpke/desktop touchups (#12)

            * Update video close button

            * Adjust movie cards to three per line 1280

            * Add PR template

            * Remove console log

            * Make nav title link

            * Address touchups for desktop

            * Create consistent margin for main image

            * Remove unused close icon

            * Do not diplsay fd or bd or slctd genre

            * Remove unused var

            * Fix movie page overview scroll and backdrop

            * Fix scrolling issues and nav jumping

            * Update favicon

            * Display runtime

            Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

            * Move PR template to .github dir

            * Place Foreign genre in correct order

            * Update movie data

            * Update Rize poster path

            Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

        commit 6f40d5b51edc8340e6ae0eaa5f5781725b1968c8
        Merge: d9ca34a a90da5e
        Author: Scott <swschipke@gmail.com>
        Date:   Thu Feb 4 20:01:19 2021 -0700

            Merge branch 'develop' of https://github.com/sschipke/black-stories-fe into develop

        commit a90da5e94fd67fff999a37e681cfe4597642099e
        Author: Scott Schipke <swschipke@gmail.com>
        Date:   Thu Feb 4 20:00:47 2021 -0700

            Add initial issue templates

            Out of the box GitHub issue templates

        * Update moviepage to delay setcurntmovie til render

        Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

* Display possible trigger warnings

* Display genres as links on movie page

* Add code of conduct page

* Update scrolling logic

* update black girl video key

Co-authored-by: Garrett Chappell <grchappell93@gmail.com>

nathanielmillard committed:
Fix gradient color.

nathanielmillard committed:
Merge branch 'main' of github.com:sschipke/black-stories-fe into previously-watched

nathanielmillard commented:
It looks like the previous watch data doesn't have the run time stored, so I am going to have some error handling to only include the run time if it's listed until we can updated the data.

nathanielmillard committed:
Add previously watched route back in.

nathanielmillard committed:
Add in conditional to control movie card render information.

nathanielmillard committed:
Insure that movies are accuratly being sorted.

nathanielmillard committed:
Make sure styling works on either movie list page.

nathanielmillard committed:
Explicitly right out undefined to handle errors.

nathanielmillard committed:
Merge develop
